modules/ROOT/pages/_partials/keycloak/8_setting-up-the-idm-sdk.adoc

```diff
@@ -0,0 +1,234 @@
+
+.Procedure
+
+1. Import the libraries
+
+[role="primary"]
+.Android
+****
+. Add the following dependency in your app's *build.gradle*:
++
+[source,groovy,subs="attributes"]
+----
+dependencies {
+    implementation "org.aerogear:android-auth:{release-number}"
+}
+----
+. To prevent build errors, add the following in your app's *build.gradle*:
+[source,groovy,subs="attributes"]
++
+----
+compileOptions {
+    sourceCompatibility 1.8
+    targetCompatibility 1.8
+}
+----
+****
+
+[role="secondary"]
+.iOS
+****
+. Add the dependency to your *Podfile*:
++
+[source,ruby,subs="attributes"]
+----
+target '[TARGET NAME]' do
+    pod 'AGSAuth', '{release-number}'
+end
+----
+
+. Update the dependencies:
++
+[source,bash]
+----
+$ pod install
+----
+
+. Import and instantiate `AGSAuth` to start using the SDK:
++
+[source,swift]
+----
+import AGSAuth
+
+auth = AGSAuth()
+----
+****
+
+[role="secondary"]
+.Cordova
+****
+Install the link:https://www.npmjs.com/package/@aerogear/auth[AeroGear Auth] package from link:https://www.npmjs.com/[NPM, window="_blank"]:
+[source,bash]
+----
+$ npm install @aerogear/auth
+----
+****
+
+[role="secondary"]
+.Xamarin
+****
+. Install link:https://docs.microsoft.com/en-us/nuget/install-nuget-client-tools[NuGet, window="_blank"].
+
+. Install the link:https://www.nuget.org/packages/AeroGear.Mobile.Core[AeroGear Core, window="_blank"] package:
++
+[source,bash,subs="attributes"]
+----
+dotnet add package AeroGear.Mobile.Core --version {release-number}
+----
+
+. For Android run:
++
+[source,bash,subs="attributes"]
+----
+dotnet add package AeroGear.Mobile.Core.Platform.Android --version {release-number}
+dotnet add package AeroGear.Mobile.Auth.Platform.Android --version {release-number}
+----
+
+. For iOS run:
++
+[source,bash,subs="attributes"]
+----
+dotnet add package AeroGear.Mobile.Core.Platform.iOS --version {release-number}
+dotnet add package AeroGear.Mobile.Auth.Platform.iOS --version {release-number}
+----
+****
+
+2. Initialize the SDK
+
+[role="primary"]
+.Android
+****
+
+. Specify the redirect URL. It is recommended to use the package name of your app.
++
+----
+AuthServiceConfiguration authServiceConfig = new AuthServiceConfiguration
+    .AuthConfigurationBuilder()
+    .withRedirectUri("org.aerogear.mobile.example:/callback")
+    .build();
+----
+
+. Create the auth service:
++
+[source,java]
+----
+AuthService authService = new AuthService(authServiceConfig);
+----
+****
+
+[role="secondary"]
+.iOS
+****
+Set your custom configuration to the auth service instance, making sure the redirect URL matches the App's Bundle Id.
+
+[source,swift]
+----
+// create the authentication config
+let authenticationConfig = AuthenticationConfig(redirectURL: "org.aerogear.mobile.example:/callback")
+try! AgsAuth.instance.configure(authConfig: authenticationConfig, useExternalUserAgent: false)
+----
+****
+
+[role="secondary"]
+.Cordova
+****
+
+Import and initialize Auth:
```

Comment:
@finp - so this is part of the initialization of the SDK "Initializing the SDK". It's not clear to me where the SDK is (it could just be me after not reading something above). 

Also, it states that I have to "initialize Auth:" with the proceeding code - I've no idea how to do this however, perhaps a Cordova dev would?

Reply:
I think cordova dev would know